packages/arb-token-bridge-ui/src/hooks/CCTP/useUpdateUsdcBalances.ts

```diff
@@ -0,0 +1,127 @@
+import { useCallback } from 'react'
+import { Address } from 'wagmi'
+import useSWRImmutable from 'swr/immutable'
+
+import { CommonAddress } from '../../util/CommonAddressUtils'
+import { getL2ERC20Address } from '../../util/TokenUtils'
+import { useNetworks } from '../useNetworks'
+import { useNetworksRelationship } from '../useNetworksRelationship'
+import { isNetwork } from '../../util/networks'
+import { useBalances } from '../useBalances'
+import { getProviderForChainId } from '@/token-bridge-sdk/utils'
+
+export async function getChildUsdcAddress({
+  parentChainId,
+  childChainId
+}: {
+  parentChainId: number
+  childChainId: number
+}) {
+  const {
+    isEthereumMainnet: isParentEthereumMainnet,
+    isSepolia: isParentSepolia
+  } = isNetwork(parentChainId)
+
+  if (isParentEthereumMainnet) {
+    return CommonAddress.ArbitrumOne.USDC
+  }
+
+  if (isParentSepolia) {
+    return CommonAddress.ArbitrumSepolia.USDC
+  }
+
+  const parentUsdcAddress = getParentUsdcAddress(parentChainId)
+  const parentProvider = getProviderForChainId(parentChainId)
+  const childProvider = getProviderForChainId(childChainId)
+
+  if (!parentUsdcAddress) {
+    return
+  }
+
+  return getL2ERC20Address({
+    erc20L1Address: parentUsdcAddress,
+    l1Provider: parentProvider,
+    l2Provider: childProvider
+  })
+}
+
+export function getParentUsdcAddress(parentChainId: number) {
```

Comment:
we can use the `CommonAddress` object directly if we use chain ids as keys instead of the names `Ethereum`, `Sepolia` etc

however such refactoring will involve many files so i'm not doing it here